### PR TITLE
Update IterEdges() documentation

### PR DESCRIPTION
**Pull request by Relux-the-Relux**
Makes clear that it does not follow the order of edges Ids addressing #830 .

### CODE REVIEW

File: networkit/graph.pyx [hunk 1]
```diff
@@ -571,6 +571,11 @@ cdef class Graph:
 	def iterEdges(self):
 		"""
 		Iterates over the edges of the graph.
+		
+		The iteration is done from node to node, iterating over all edges
+		of a node before passing to edges of the next node in the order of node Ids.
```
Comment (angriman) [inline networkit/graph.pyx:576]: Maybe rephrase with "For each node u in the graph in ascending node id order, the iterator yields the edges incident to u".

**Comment by avdgrinten** [comment]
In addition to this change, we should update the docs of the edge centrality algos to state that their result is indexed by edge ID.

**Comment by avdgrinten** [comment]
This is also underspecified for undirected graphs. Which direction of an edge is returned?

**Comment by angriman** [comment]
Yes, we should specify that for undirected graphs we return edges `(u, v)` where `u < v` (to avoid iterating twice over the same edge) while for directed graphs we return out-edges.

File: networkit/graph.pyx [hunk 2]
```diff
@@ -571,6 +571,12 @@ cdef class Graph:
 	def iterEdges(self):
 		"""
 		Iterates over the edges of the graph.
+		
+		For each node u in the graph in ascending node id order,
+		the iterator yields the out-edges of u in directed Graphs
+		and the edges (u,v) in which u < v for undirected Graphs.
```
Comment (angriman) [inline networkit/graph.pyx:577]: Graphs -> graphs

**Review by angriman — APPROVED** [review]

File: networkit/centrality.pyx [hunk 3]
```diff
@@ -129,7 +129,8 @@ cdef class Betweenness(Centrality):
 
 
 	def edgeScores(self):
-		""" Get a vector containing the betweenness score for each edge in the graph.
+		""" Get a vector containing the betweenness score for each edge in the graph
+		in ascending edge id order.
```
Comment (avdgrinten) [inline networkit/centrality.pyx:133]: `edge ID`, not `id`.

Comment (avdgrinten) [reply networkit/centrality.pyx:133]: Maybe also add `(as returned by Graph.edgeId())`.

File: networkit/graph.pyx [hunk 4]
```diff
@@ -571,6 +571,12 @@ cdef class Graph:
 	def iterEdges(self):
 		"""
 		Iterates over the edges of the graph.
+		
+		For each node u in the graph in ascending node id order,
```
Comment (avdgrinten) [inline networkit/graph.pyx:575]: `id` -> `ID`.

File: networkit/graph.pyx [hunk 5]
```diff
@@ -571,6 +571,12 @@ cdef class Graph:
...
+		the iterator yields the out-edges of u in directed graphs
+		and the edges (u,v) in which u < v for undirected graphs.
+		
+		It does NOT follow the order of edgeIds (if present).
```
Comment (avdgrinten) [inline networkit/graph.pyx:579]: Do not capitalize `NOT`. `edgeIds` -> `edge IDs`.

**Review by avdgrinten — CHANGES_REQUESTED** [review]

File: networkit/centrality.pyx (hunk 3 above)
Comment (angriman) [reply networkit/centrality.pyx]: @Relux-the-Relux this comment is yet to be addressed.

**Review by avdgrinten — APPROVED** [review]
LGTM.
